memorystore/valkey/session/GUIDE.md

```diff
@@ -0,0 +1,202 @@
+
+- **Spring Boot Application:** Manages session logic and provides APIs for interaction.
+- **Valkey (In-Memory Cache):** Stores active session data for fast lookups.
+- **PostgreSQL Database:** Acts as the persistent storage for session data.
+- **Google Cloud Platform Services:** Hosts the application and its dependencies.
+
+## Step-by-Step Guide
+
+To begin, we are will generate an API with the following routes:
+
+1. **Session Creation:** When a user logs in, a session is created and stored in the database cache with a Time-To-Live (TTL).
+2. **Session Retrieval:** For every request, the session is checked in the cache. If not found, it is retrieved from the database and re-cached.
+3. **Session Update:** When a session is updated (e.g., extended TTL), both the cache and database are updated.
+4. **Session Invalidation:** When a user logs out or the session expires, it is removed from the cache and database.
+
+### Creating a new application
+
+The first step is to initialize a Spring Boot application. The [official guide](https://spring.io/guides/gs/spring-boot) demonstrates how to generate a new project using [Spring Initializer](https://start.spring.io/).
+
+1. Choose `Maven` as the project type for this demonstration.
+2. Select Sprint Boot version 3.4.1.
+3. Complete the appropriate metadata.
+4. Choose your preferred `Packing` for downloading.
+5. Select `Java 17` for your Java version.
+6. Finally, generate and extract the files.
+
+#### Installing additional dependencies
+
+Next, ensure the following dependencies have been added to your POM.xml file.
+
+#### Jedis
+
+Add the folowing snippet toconnect directly to the Memorystore for Valkey instance.
+
+```xml
+<!-- Jedis: Redis Java Client -->
+<dependency>
+   <groupId>redis.clients</groupId>
+   <artifactId>jedis</artifactId>
+   <version>4.3.0</version>
+</dependency>
+```
+
+#### Jakarta
+
+To ensure that our API routes are correctly validated. Add the following dependency.
+This enables the use of annotations like `@NotNull` and `@Size` on classes to automatically enforce input constraints, reducing the need for manual validation logic.
+
+```xml
+<!-- Add Validation support-->
+<dependency>
+   <groupId>jakarta.validation</groupId>
+   <artifactId>jakarta.validation-api</artifactId>
+   <version>3.0.2</version>
+</dependency>
+```
+
+### Connecting the Service layer
+
+Next, add the following to route logic to the API.
+
+#### Logging in a user
+
+```java
+  public String login(String username, String password) {
+    // Authenticate user
+    Optional<Integer> userId = accountRepository.authenticateUser(
+      username,
+      password
+    );
+
+    // No user found
+    if (userId.isEmpty()) {
+      return null;
+    }
+
+    // Generate token for the user
+    String token = Utils.generateToken(Global.TOKEN_BYTE_LENGTH);
+
+    try {
+      jedis.set(token, username);
+      jedis.expire(token, Global.TOKEN_EXPIRATION);
+    } catch (Exception e) {
```

Comment:
so these jedis methods don't typically throw checked exceptions (per https://www.javadoc.io/doc/redis.clients/jedis/3.0.1/redis/clients/jedis/Jedis.html#setex-java.lang.String-int-java.lang.String-) none of them do.

maybe just checking return codes makes sense & don't worry about catching 'Exception'.

if you want you could catch RuntimeException & retrhow with a message like 'failed to store session' etc...)